source/contentRecog/recogUi.py

```diff
@@ -0,0 +1,133 @@
+	name = _("Result")
+	treeInterceptor = None
+
+	def __init__(self, result=None, obj=None):
+		self.parent = parent = api.getFocusObject()
+		self.result = result
+		self._selection = self.makeTextInfo(textInfos.POSITION_FIRST)
+		super(RecogResultNVDAObject, self).__init__(windowHandle=parent.windowHandle)
+
+	def makeTextInfo(self, position):
+		# Maintain our own fake selection/caret.
+		if position == textInfos.POSITION_SELECTION:
+			ti = self._selection.copy()
+		elif position == textInfos.POSITION_CARET:
+			ti = self._selection.copy()
+			ti.collapse()
+		else:
+			ti = self.result.makeTextInfo(self, position)
+		return self._patchTextInfo(ti)
+
+	def _patchTextInfo(self, info):
+		# Patch TextInfos so that updateSelection/Caret updates our fake selection.
+		info.updateCaret = lambda: self._setSelection(info, True)
+		info.updateSelection = lambda: self._setSelection(info, False)
+		# Ensure any copies get patched too.
+		oldCopy = info.copy
+		info.copy = lambda: self._patchTextInfo(oldCopy())
+		return info
+
+	def _setSelection(self, textInfo, collapse):
+		self._selection = textInfo.copy()
+		if collapse:
+			self._selection.collapse()
+
+	def setFocus(self):
+		ti = self.parent.treeInterceptor
+		if isinstance(ti, browseMode.BrowseModeDocumentTreeInterceptor):
+			# Normally, when entering browse mode from a descendant (e.g. dialog),
+			# we want the cursor to move to the focus (#3145).
+			# However, we don't want this for recognition results, as these aren't focusable.
+			ti._enteringFromOutside = True
+		eventHandler.executeEvent("gainFocus", self)
+
+	def script_activatePosition(self, gesture):
+		try:
+			self._selection.activate()
+		except NotImplementedError:
+			log.debugWarning("Result TextInfo does not implement activate")
+	# Translators: Describes a command.
+	script_activatePosition.__doc__ = _("Activates the text at the cursor if possible")
+
+	def script_exit(self, gesture):
+		eventHandler.executeEvent("gainFocus", self.parent)
+	# Translators: Describes a command.
+	script_exit.__doc__ = _("Dismiss the recognition result")
+
+	__gestures = {
+		"kb:enter": "activatePosition",
+		"kb:escape": "exit",
+	}
+
+#: Keeps track of the recognition in progress, if any.
+_activeRecog = None
+def recognizeNavigatorObject(recognizer):
+	"""User interface function to recognize content in the navigator object.
+	This should be called from a script or in response to a GUI action.
+	@param recognizer: The content recognizer ot use.
+	@type recognizer: L{contentRecog.ContentRecognizer}
+	"""
+	global _activeRecog
+	if _activeRecog:
+		_activeRecog.cancel()
+	# Translators: Reporting when content recognition (e.g. OCR) begins.
+	ui.message(_("Recognizing"))
+	nav = api.getNavigatorObject()
+	left, top, width, height = nav.location
+	resize = recognizer.getResizeFactor(width, height)
+	coordConv = ResultCoordConverter(left, top, resize)
+	destWidth = int(width * resize)

```

Comment:
I think it would be nice if the caller did not need to know about which direction the resize works, or the offsets with the resultCoordConverter.